sdk/core/core/inc/az_http_response_parser.h

```diff
@@ -90,6 +90,14 @@ az_http_response_get_next_header(az_span const self, az_pair * const p_header);
 AZ_NODISCARD az_result
 az_http_response_get_body(az_span const self, az_pair * const p_last_header, az_span * const body);
 
+/**
+ * Get an HTTP header by name.
+ */
+AZ_NODISCARD az_result az_http_response_get_header_by_name(
+    az_span const self,
```

Comment:
can't read `self` without thinking about python...  XD

Reply:
Smalltalk, Object Pascal, Perl, Python, Ruby, Rust, Objective-C, DataFlex, Swift, and others ;-)

Reply:
Similar to other comment i made this morning, self does not indicate what the span contents must be.
I'd actually like a typedef for the http response type and then all methods that require an http_reponse should have this typedef as the first parameter so that there is type safety at compile time here. An http response is NOT just a span; it is a span whose content is in a specific format; we need a type definition to reflect this.